### PR TITLE
Add port Deltarune

### PR DESCRIPTION
**Pull request by JeodC**
# New Port for Deltarune

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/deltarune/Deltarune.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export PATCHER_TIME="2 to 5 minutes"
+
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_linux"
+
+# CD and set permissions
+cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
```
Comment (JanTrueno) [inline ports/deltarune/Deltarune.sh:34]: Why do all files in gamedir need executable permissions?

File: ports/deltarune/port.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  "items_opt": [],
+  "attr": {
+    "title": "Deltarune",
+    "porter": [
+      "Jeod"
+    ],
+    "desc": "In this sequel to Undertale, Kris and Susie are drawn into a mysterious Dark World, where they join forces with Ralsei to close the Dark Fountains and restore balance. Along the way, they encounter strange foes, form unlikely friendships, and uncover new mysteries tied to the world of monsters.",
+    "desc_md": null,
+    "inst": "Download the Windows version from https://tobyfox.itch.io/deltarune and unzip to `ports/deltarune/gamedata`.",
```
Comment (JanTrueno) [inline ports/deltarune/port.json:16]: Is the game always shipped as a zip? Could be nice to add an auto-extractor.

File: ports/deltarune/Deltarune.sh (hunk 1 above)
Comment (JeodC) [reply ports/deltarune/Deltarune.sh:34]: Because it's a standard now.

File: ports/deltarune/port.json (hunk 2 above)
Comment (JeodC) [reply ports/deltarune/port.json:16]: Could be, but don't need to.

**Review by JanTrueno — APPROVED** [review]

File: ports/deltarune/deltarune/libs/libcrypto.so.1.1
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libcrypto.so.1.1:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libcrypto.txt

File: ports/deltarune/deltarune/libs/libFLAC.so.8
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libFLAC.so.8:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libflac.txt

File: ports/deltarune/deltarune/libs/libogg.so.0
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libogg.so.0:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libogg.txt

File: ports/deltarune/deltarune/libs/libopenal.so.1
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libopenal.so.1:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libopenal.txt

File: ports/deltarune/deltarune/libs/libvorbis.so.0
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libvorbis.so.0:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libvorbis.txt

File: ports/deltarune/deltarune/libs/libvorbisenc.so.2
Comment (cdeletre) [inline ports/deltarune/deltarune/libs/libvorbisenc.so.2:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libvorbisenc.txt

File: ports/deltarune/deltarune/libs32/libzip.so.5
Comment (cdeletre) [inline ports/deltarune/deltarune/libs32/libzip.so.5:1]: https://github.com/cdeletre/PortMaster-MiniDoom2/blob/main/minidoom2/LICENSE.libzip.txt

**Review by cdeletre — CHANGES_REQUESTED** [review]
I think it's good if we can get used to include the licenses for the libraries used in a port. I know it's not a fun administrative task :)

**Comment by JeodC** [comment]
> I think it's good if we can get used to include the licenses for the libraries used in a port. I know it's not a fun administrative task :)

This has not been discussed or standardized yet afaik, and I'd think this would only apply to libraries built from source, not redistributed.

**Comment by cdeletre** [comment]
> I think it's good if we can get used to include the licenses for the libraries used in a port. I know it's not a fun administrative task :)



> > I think it's good if we can get used to include the licenses for the libraries used in a port. I know it's not a fun administrative task :)
> 
> This has not been discussed or standardized yet afaik, and I'd think this would only apply to libraries built from source, not redistributed.

You're right, I haven't thought about discussing this before asking for it in the last PRs I've reviewed. Let's leave it as is for now, we can always come back on this later if needed.

**Review by cdeletre — APPROVED** [review]
✅ good to go

**Pull request merged**